src/core/qgstemporalnavigationobject.cpp

```diff
@@ -78,6 +78,7 @@ QgsExpressionContextScope *QgsTemporalNavigationObject::createExpressionContextS
   scope->setVariable( QStringLiteral( "frame_duration" ), mFrameDuration, true );
   scope->setVariable( QStringLiteral( "frame_timestep" ), mFrameDuration.originalDuration(), true );
   scope->setVariable( QStringLiteral( "frame_timestep_unit" ), mFrameDuration.originalUnit(), true );
+  scope->setVariable( QStringLiteral( "frame_timestep_units" ), QgsUnitTypes::toString( mFrameDuration.originalUnit() ), true );
```

Comment:
QgsExpressionContext::filteredVariableNames should now strip out the older frame_timestep_unit variable

Reply:
@nyalldawson I'm not sure how do you want to achieve this, If I add `_` to the start of `frame_timestep_unit` it will be filtered out but the tests coverage for it in `TestQgsTemporalNavigationObject::expressionContext()` will fail!

Reply:
I'd just hardcode "frame_timestep_unit" as an new exception in QgsExpressionContext::filteredVariableNames.

Reply:
> I'd just hardcode "frame_timestep_unit" as an new exception in QgsExpressionContext::filteredVariableNames.

Right!

Reply:
@nyalldawson I have provided the change as you requested, I ll be more happy if we had a more robust way for removing variable names when filtering them. I'm planning on making another PR for that if you are okay with that. thanks.